modules/c++/math.linear/include/math/linear/MatrixMxN.h

```diff
@@ -43,10 +43,30 @@ template<typename _T> inline bool equals(const _T& e1, const _T& e2, _T eps)
 
 template<> inline bool equals(const float& e1, const float& e2)
 {
+    if (e1 == std::numeric_limits<float>::infinity() &&
```

Comment:
Rather than this, shouldn't the base `equals()` method that takes in an `eps` be updated to do this?

To confirm, if you say
```
float e1 = std::numeric_limits<float>::infinity();
float e2 = std::numeric_limits<float>::infinity();
bool val =  e1 == e2;
```

`val` will be true right?  If so, I agree that we should add this logic.